ibtk/src/math/PoissonUtilities.cpp

```diff
@@ -994,32 +628,31 @@ void PoissonUtilities::computeSCMatrixCoefficients(Pointer<Patch<NDIM> > patch,
         }
     }
     return;
-} // computeSCMatrixCoefficients
-
-void PoissonUtilities::adjustCCBoundaryRhsEntries(Pointer<Patch<NDIM> > patch,
-                                                  CellData<NDIM, double>& rhs_data,
-                                                  const PoissonSpecifications& poisson_spec,
-                                                  RobinBcCoefStrategy<NDIM>* bc_coef,
-                                                  double data_time,
-                                                  bool homogeneous_bc)
+}
+
+void PoissonUtilities::adjustRHSAtPhysicalBoundary(CellData<NDIM, double>& rhs_data,
+                                                   Pointer<Patch<NDIM> > patch,
+                                                   const PoissonSpecifications& poisson_spec,
+                                                   RobinBcCoefStrategy<NDIM>* bc_coef,
+                                                   double data_time,
+                                                   bool homogeneous_bc)
 {
-    adjustCCBoundaryRhsEntries(
-        patch, rhs_data, poisson_spec, std::vector<RobinBcCoefStrategy<NDIM>*>(1, bc_coef), data_time, homogeneous_bc);
+    std::vector<RobinBcCoefStrategy<NDIM>*> bc_coefs(1, bc_coef);
+    adjustRHSAtPhysicalBoundary(rhs_data, patch, poisson_spec, bc_coefs, data_time, homogeneous_bc);
     return;
-} // adjustCCBoundaryRhsEntries
-
-void PoissonUtilities::adjustCCBoundaryRhsEntries(Pointer<Patch<NDIM> > patch,
-                                                  CellData<NDIM, double>& rhs_data,
-                                                  const PoissonSpecifications& poisson_spec,
-                                                  const std::vector<RobinBcCoefStrategy<NDIM>*>& bc_coefs,
-                                                  double data_time,
-                                                  bool homogeneous_bc)
+}
+
+void PoissonUtilities::adjustRHSAtPhysicalBoundary(CellData<NDIM, double>& rhs_data,
+                                                   Pointer<Patch<NDIM> > patch,
+                                                   const PoissonSpecifications& poisson_spec,
+                                                   const std::vector<RobinBcCoefStrategy<NDIM>*>& bc_coefs,
+                                                   double data_time,
+                                                   bool homogeneous_bc)
 {
-    const int depth = static_cast<int>(bc_coefs.size());
+    const int depth = rhs_data.getDepth();
 #if !defined(NDEBUG)
-    TBOX_ASSERT(rhs_data.getDepth() == depth);
+    TBOX_ASSERT((int)bc_coefs.size() == depth);
```

Comment:
Yep, that's the right thing to do, but C casts are so succinct to use. I will change it. I almost first use static_cast<> and then I realize how ugly they look and then covert it to C cast :-)